src/eei.cpp

```diff
@@ -335,16 +335,16 @@ string toHex(evm_uint256be const& value) {
 
       HERA_DEBUG << "externalCodeCopy " << hex << addressOffset << " " << resultOffset << " " << codeOffset << " " << length << dec << "\n";
 
-      evm_address address = loadUint160(addressOffset);
-      const uint8_t *code;
-      size_t code_size = context->fn_table->get_code(&code, context, &address);
+      evmc_address address = loadUint160(addressOffset);
+      vector<uint8_t> codeBuffer(length);
+      size_t numCopied = context->fn_table->copy_code(context, &address, codeOffset, codeBuffer.data(), codeBuffer.size());
+      fill_n(&codeBuffer[numCopied], length - numCopied, 0);
 
       heraAssert(ffs(GasSchedule::copy) + (ffs(length) - 5) <= 64, "Gas charge overflow");
       heraAssert(numeric_limits<uint64_t>::max() - GasSchedule::extcode >= GasSchedule::copy * ((uint64_t(length) + 31) / 32), "Gas charge overflow");
       takeGas(GasSchedule::extcode + GasSchedule::copy * ((uint64_t(length) + 31) / 32));
-      // NOTE: code will be freed by the callee (client)
       // FIXME: optimise this so not vector needs to be created
```

Comment:
Comment is invalid now.

Reply:
It is still valid. You can copy the code directly to the WASM memory.

Reply:
But then it should be moved to the point where the vector is created?